sky/serve/infra_providers.py

```diff
@@ -622,10 +637,35 @@ def _probe_replica(info: ReplicaInfo) -> Tuple[str, bool]:
             if info.first_not_ready_time is None:
                 info.first_not_ready_time = time.time()
             if info.status_property.service_once_ready:
+                if self.use_spot:
+                    # Pull the actual cluster status from the cloud provider to
+                    # determine whether the cluster is preempted.
+                    (cluster_status, handle
+                    ) = backends.backend_utils.refresh_cluster_status_handle(
+                        cluster_name,
+                        force_refresh_statuses=set(status_lib.ClusterStatus))
+
+                    if cluster_status != status_lib.ClusterStatus.UP:
+                        # The cluster is (partially) preempted. It can be down, INIT
+                        # or STOPPED, based on the interruption behavior of the cloud.
+                        # Spot recovery is needed.
+                        cluster_status_str = (
+                            '' if cluster_status is None else
+                            f' (status: {cluster_status.value})')
+                        logger.info(
+                            f'Cluster {cluster_name} is preempted: {cluster_status_str}.'
```

Comment:
```suggestion
                            f'Cluster {cluster_name} is preempted{cluster_status_str}.'
```
To better handle the case when `cluster_status is None` ;)

Reply:
Done, thanks.